cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,411 @@
+                                                   'update', 'delete',
+                                                   'expand']),
+            verify_ssl_cert=dict(required=False, default=True, type='bool'),
+            wait=dict(required=False, default=False, type='bool'),
+            wait_time=dict(required=False, default=600, type='int'),
+            wait_poll_interval=dict(required=False, default=2, type='int')
+        )
+    )
+
+    if not HAS_LIBCLOUD:
+        module.fail_json(msg='libcloud is required for this module.')
+
+    # set short vars for readability
+    credentials = get_credentials()
+    if credentials is False:
+        module.fail_json(msg="User credentials not found")
+    user_id = credentials['user_id']
+    key = credentials['key']
+    region = 'dd-%s' % module.params['region']
+    location = module.params['location']
+    network_domain_name = module.params['network_domain']
+    name = module.params['name']
+    vlan_id = module.params['vlan_id']
+    description = module.params['description']
+    base_address = module.params['private_ipv4_base_address']
+    prefix_size = module.params['private_ipv4_prefix_size']
+    verify_ssl_cert = module.params['verify_ssl_cert']
+    action = module.params['action']
+
+    # Instantiate driver
+    libcloud.security.VERIFY_SSL_CERT = verify_ssl_cert
+    DimensionData = get_driver(Provider.DIMENSIONDATA)
+    driver = DimensionData(user_id, key, region=region)
+
+    # Get Network Domain Object
+    network_domain = get_network_domain_by_name(driver, network_domain_name,
+                                                location)
+
+    # Process action
+    if action == 'create':
+        if name is False:
+            module.fail_json(msg="'name' is a required argument when action" +
+                                 " is 'create'")
+        res = create_vlan(module, driver, location, network_domain, name,
+                          description, base_address, prefix_size)
+        vlan = vlan_obj_to_dict(res['vlan'])
+        if res['changed'] is False:
+            module.exit_json(changed=False, msg="VLAN with name '%s'" +
+                             " already exists." % name, vlan=vlan)
+        module.exit_json(changed=True, msg="Successfully created VLAN.",
+                         vlan=vlan)
+    elif action == 'read' or action == 'get':
+        res = get_vlan(module, driver, location, network_domain,
+                       vlan_id, name)
+        if type(res) is dict:
+            vlan = res
+        else:
+            vlan = vlan_obj_to_dict(res)
+        module.exit_json(changed=False, msg="Successfully read VLAN.",
+                         vlan=vlan)
+    elif action == 'delete':
+        res = delete_vlan(module, driver, location, network_domain,
+                          vlan_id, name)
+        module.exit_json(changed=res, msg="Successfully deleted VLAN or " +
+                         "VLAN does not exist")
```

Comment:
Note: python combines literal strings that are next to each other.  So this can just be written as:
``` python
+        module.exit_json(changed=res, msg="Successfully deleted VLAN or "
+                         "VLAN does not exist")
```